File: OC.Domain/Models/Branches/CellLeadership.cs
﻿using OC.Domain.Interfaces;
using OC.Domain.Models.Members;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace OC.Domain.Models.Branches
{
    public class CellLeadership : AuditableEntity, IEntityBase
    {
        public long Id { get; set; }
        public long CellGroupId { get; set; }
        public long LeadershipRoleId { get; set; }
        public long MemberId { get; set; }

        // Relationships
        public virtual CellGroup CellGroup { get; set; }
        public virtual LeadershipRole LeadershipRole { get; set; }
        public virtual Member Member { get; set; }
    }
}

File: OC.UI/src/app/blci/api/branches/cell-leadership.ts
import { LeadershipRole } from "../members/leadership-role";
import { Member } from "../members/member";
import { CellGroup } from "./cell-group";

export interface CellLeadership {
    id: number;
    cellGroupId: number;
    leadershipRoleId: number;
    memberId: number;
    isActive: boolean;
    cellGroup: CellGroup;
    leadershipRole: LeadershipRole;
    member: Member;
}
